openfeature-provider-local/src/main/java/com/spotify/confidence/SwapWasmResolverApi.java

```diff
@@ -1,37 +1,184 @@
     this.secondaryWasmResolverApi = new WasmResolveApi(flagLogger);
-    this.secondaryWasmResolverApi.setResolverState(initialState);
+    this.secondaryWasmResolverApi.setResolverState(initialState, accountId);
     this.wasmResolverApiRef.set(primaryWasmResolverApi);
   }
 
-  public void updateState(byte[] state) {
+  public void updateState(byte[] state, String accountId) {
     if (isPrimary) {
-      this.secondaryWasmResolverApi.setResolverState(state);
+      this.secondaryWasmResolverApi.setResolverState(state, accountId);
       this.wasmResolverApiRef.set(secondaryWasmResolverApi);
     } else {
-      this.primaryWasmResolverApi.setResolverState(state);
+      this.primaryWasmResolverApi.setResolverState(state, accountId);
       this.wasmResolverApiRef.set(primaryWasmResolverApi);
     }
     isPrimary = !isPrimary;
   }
 
+  public void close() {
+    primaryWasmResolverApi.close();
+    secondaryWasmResolverApi.close();
+  }
+
   private final ReentrantLock logResolveLock = new ReentrantLock();
 
+  public CompletableFuture<ResolveFlagsResponse> resolveWithSticky(
+      ResolveWithStickyRequest request) {
+    logResolveLock.lock();
+    final var response = resolveWithStickyInternal(request);
+    logResolveLock.unlock();
+    return response;
+  }
+
+  private CompletableFuture<ResolveFlagsResponse> resolveWithStickyInternal(
+      ResolveWithStickyRequest request) {
+    final var response = wasmResolverApiRef.get().resolveWithSticky(request);
+
+    switch (response.getResolveResultCase()) {
+      case SUCCESS -> {
+        final var success = response.getSuccess();
+        // Store updates if present
+        if (!success.getUpdatesList().isEmpty()) {
+          storeUpdates(success.getUpdatesList());
+        }
+        return CompletableFuture.completedFuture(success.getResponse());
+      }
+      case MISSING_MATERIALIZATIONS -> {
+        final var missingMaterializations = response.getMissingMaterializations();
+
+        // Check for ResolverFallback first - return early if so
+        if (stickyResolveStrategy instanceof ResolverFallback fallback) {
+          return fallback.resolve(request.getResolveRequest());
+        }
+
+        // Handle MaterializationRepository case
+        if (stickyResolveStrategy instanceof MaterializationRepository repository) {
+          final var currentRequest =
+              handleMissingMaterializations(
+                  request, missingMaterializations.getItemsList(), repository);
+          return resolveWithStickyInternal(currentRequest);
+        }
+
+        throw new RuntimeException(
+            "Unknown sticky resolve strategy: " + stickyResolveStrategy.getClass());
+      }
+      case RESOLVERESULT_NOT_SET ->
+          throw new RuntimeException("Invalid response: resolve result not set");
+      default ->
+          throw new RuntimeException("Unhandled response case: " + response.getResolveResultCase());
```

Comment:
What happens to the final evaluation returned to the application, and subsequent ones? Is this turning into a default value with error code on the openfeature side? I am thinking some integration tests would be great in future PRs